src/core/search/indices.cc

```diff
@@ -71,19 +71,21 @@ absl::flat_hash_set<string> NormalizeTags(string_view taglist, bool case_sensiti
 NumericIndex::NumericIndex(PMR_NS::memory_resource* mr) : entries_{mr} {
 }
 
+bool NumericIndex::Matches(DocId id, DocumentAccessor* doc, string_view field) {
+  auto strings = doc->GetStrings(field);
+  return std::all_of(strings.begin(), strings.end(),
+                     [](const auto& str) { return ParseNumericField(str); });
+}
+
 void NumericIndex::Add(DocId id, DocumentAccessor* doc, string_view field) {
   for (auto str : doc->GetStrings(field)) {
-    double num;
-    if (absl::SimpleAtod(str, &num))
-      entries_.emplace(num, id);
+    entries_.emplace(ConvertToDouble(str), id);
```

Comment:
the assumption here is that all values are convertible to double? why is that?

Reply:
Before it was matched => proved that all fields are doubles